main.py

```diff
@@ -0,0 +1,365 @@
+import requests
+import json
+import datetime
+import os
+from operator import itemgetter
+
+
+# GLOBALS
+PATH = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))   # Gets the absolute path for the running file
+FILE_NAME = "Riven_data_{platform}_{monday:%y}-{monday:%m}-{monday:%d}.json"
+
+RAW_PATH = os.path.join(PATH, 'data/{platform}/raw')
+RAW_FILE_PATH = os.path.join(RAW_PATH, FILE_NAME)
+
+EDIT_PATH = os.path.join(PATH, "data/{platform}/edited")
+TOTAL_FILE_PATH = os.path.join(EDIT_PATH, "TOTAL_{platform}.json")
+
+L_MONDAY = ""
+PLATFORMS = ["PC", "PS4", "XB1", "SWI"]
+URL = "https://n9e5v4d8.ssl.hwcdn.net/repos/weeklyRivens{}.json"
+
+
+def last_monday():
+    '''Gets the last monday date'''
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday())
+    return monday
+
+
+def get_riven_json(platform):
+    '''Gets the json for a given platform'''
+    print(f"Checking if this week json for {platform} is saved!")
+    file_path = os.path.normpath(RAW_FILE_PATH.format(platform=platform, monday=L_MONDAY))
+
+    # Checks if the file already exists
+    if os.path.isfile(file_path):
+        print("File already exists!\n-----------------")
+        return
+
+    print("No file found, creating it!\n-----------------")
+    r = requests.get(URL.format(platform))
+    r.raise_for_status()
+    data = r.json()
+    with open(file_path, "w") as outfile:
+        json.dump(data, outfile, indent=4)
+
+
+def process_data(platform):
+    '''Calls all of the processing functions for every raw file'''
+
+    # Goes over every file in the raw folder and processes the data for it
+    for filename in os.listdir(os.path.normpath(RAW_PATH.format(platform=platform))):
+
+        print(f"\nProcessing raw {filename}\n")
+
+        date = filename.rstrip(".json")[-8:]
+        raw_file_path = os.path.normpath(os.path.join(RAW_PATH.format(platform=platform), filename))
+        edit_file_path = os.path.normpath(os.path.join(EDIT_PATH.format(platform=platform), filename))
+
+        with open(raw_file_path, "r") as raw:
+            raw_data = json.load(raw)
+            sales = sale_calc(raw_data)
+            raw_dict = create_dict(raw_data, 1)
+            del raw_data
+
+        # Calls total_json which handles the total file and returns the total_dict
+        total_dict = total_json(platform, raw_dict, date, sales)
+
+        if os.path.isfile(edit_file_path):
+            print(f"/edited/{filename} Aleady exists!")
+        else:
+            comparison(platform, raw_dict, total_dict, sales, edit_file_path)
+
+
+def sale_calc(data):
```

Comment:
Yes `sale_calc` looks for the lowest popularity riven and checks if the max and min price are the same (you only need those 2 since if those are the same average, stddev and median will also be the same).
I will think about how to handle the unlikely scenario. Not having the total amount of sales is really problematic since we have to assume a lot of things that we can't really confirm.